packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -296,51 +190,52 @@ contract ControllerHelper is UniswapControllerHelper, AaveControllerHelper, IERC
+    function closeShortWithUserNft(ControllerHelperDataType.CloseShortWithUserNftParams calldata _params) external {
         INonfungiblePositionManager(nonfungiblePositionManager).safeTransferFrom(
             msg.sender,
             address(this),
             _params.tokenId
         );
 
-        _closeUniLp(
-            closeUniLpParams({
-                vaultId: _params.vaultId,
+        (uint256 wPowerPerpAmountInLp, ) = ControllerHelperUtil.closeUniLp(
+            nonfungiblePositionManager,
+            ControllerHelperDataType.closeUniLpParams({
                 tokenId: _params.tokenId,
                 liquidity: _params.liquidity,
                 liquidityPercentage: _params.liquidityPercentage,
-                wPowerPerpAmountToBurn: _params.wPowerPerpAmountToBurn,
-                collateralToWithdraw: _params.collateralToWithdraw,
-                limitPriceEthPerPowerPerp: _params.limitPriceEthPerPowerPerp,
                 amount0Min: _params.amount0Min,
                 amount1Min: _params.amount1Min
-            })
+            }),
+            isWethToken0
+        );
+        _closeShortWithAmountsFromLp(
+            _params.vaultId,
+            wPowerPerpAmountInLp,
+            _params.wPowerPerpAmountToBurn,
+            _params.collateralToWithdraw,
+            _params.limitPriceEthPerPowerPerp
+        );
+        ControllerHelperUtil.checkPartialLpClose(
+            nonfungiblePositionManager,
+            controller,
+            0,
+            _params.tokenId,
+            _params.liquidityPercentage
         );
-
-        _checkPartialLpClose(0, _params.tokenId, _params.liquidityPercentage);
 
         IWETH9(weth).withdraw(IWETH9(weth).balanceOf(address(this)));
         payable(msg.sender).sendValue(address(this).balance);
     }
 
-    function flashloanCloseVaultLpNft(FlashloanCloseVaultLpNftParam calldata _params) external payable {
+    function flashloanCloseVaultLpNft(ControllerHelperDataType.FlashloanCloseVaultLpNftParam calldata _params)
+        external
+        payable
+    {
         _flashLoan(
             weth,
             _params.collateralToFlashloan,
-            uint8(CALLBACK_SOURCE.FLASHLOAN_CLOSE_VAULT_LP_NFT),
-            abi.encode(
-                closeUniLpParams({
-                    vaultId: _params.vaultId,
-                    tokenId: _params.tokenId,
-                    liquidity: _params.liquidity,
-                    liquidityPercentage: _params.liquidityPercentage,
-                    wPowerPerpAmountToBurn: _params.wPowerPerpAmountToBurn,
-                    collateralToWithdraw: _params.collateralToFlashloan.add(_params.collateralToWithdraw),
-                    limitPriceEthPerPowerPerp: _params.limitPriceEthPerPowerPerp,
-                    amount0Min: _params.amount0Min,
-                    amount1Min: _params.amount1Min
-                })
-            )
+            uint8(ControllerHelperDataType.CALLBACK_SOURCE.FLASHLOAN_CLOSE_VAULT_LP_NFT),
```

Comment:
I feel like this and FLASHLOAN_REBALANCE_VAULT_NFT have some repeated functionality around closing an LP